plugins/rikki/heroeslounge/components/SlothAccount.php

```diff
@@ -461,6 +460,39 @@ public function onUpdateLinks()
         }
     }
 
+    public function onSyncDiscord()
+    {
+        if (!$user = $this->user()) {
+            return;
+        }
+
+        $sloth = SlothModel::getFromUser($user);
+
+        if (!empty($sloth->discord_id)) {
+            $newTag = Discord\Attendance::getDiscordTag($sloth->discord_id);
+
+            if (!empty($newTag)) {
+                $sloth->discord_tag = $newTag;
+                $sloth->save();
+    
+                Flash::success('Discord tag was updated successfully!');
+                if ($redirect = $this->makeRedirection()) {
+                    return $redirect;
+                }
+            } else {
+                Flash::error(`Could not update Discord tag!`);
+                if ($redirect = $this->makeRedirection()) {
+                    return $redirect;
+                }
+            }
+        } else {
+            Flash::error(`Could not update Discord tag, please contact a moderator for assistance.`);
```

Comment:
Please allow users who do not have a valid id to just enter their Discord tag on their own. We do not want additional workload for mods when aviodable.

Reply:
I feel like this will cause confusion for users that will instead try to manually enter their Discordtag instead of having us do the update. I'm quite certain that you're also aware how much our users already struggle getting their new tag right on registration, let alone if they try so here, without verification.

Reply:
I trust users are able to enter their own discord tag. Feel free to verify it and display a message if they are not found on the server. If all fails, mods can still help out.